File: app/src/main/java/com/apicloud/pkg/sdk/mvp/presenter/MainPresenter.java
package com.apicloud.pkg.sdk.mvp.presenter;


import android.content.Context;

import javax.inject.Inject;

import com.apicloud.pkg.sdk.api.AppApi;
import com.apicloud.pkg.sdk.common.RxPresenter;
import com.apicloud.pkg.sdk.mvp.contract.MainContract;
import com.apicloud.pkg.sdk.databinding.ActivityMainBinding;

public class MainPresenter extends RxPresenter<MainContract.View> implements MainContract.Presenter {

    private AppApi appApi;
    private Context context;
    private ActivityMainBinding binding;

    @Inject
    public MainPresenter(AppApi appApi, Context context, ActivityMainBinding binding) {
        super();
        this.appApi = appApi;
        this.context = context;
        this.binding = binding;
    }

    public void toInit() {
        binding.setMainModel(new MainModel());
    }

    public class MainModel {

    }

}

File: app/src/main/java/com/apicloud/pkg/sdk/mvp/presenter/MainPresenter.kt
package com.apicloud.pkg.sdk.mvp.presenter


import android.content.Context
import com.apicloud.pkg.sdk.api.AppApi
import com.apicloud.pkg.sdk.common.RxPresenter
import com.apicloud.pkg.sdk.data.UserInfo
import com.apicloud.pkg.sdk.databinding.ActivityMainBinding
import com.apicloud.pkg.sdk.mvp.contract.MainContract
import javax.inject.Inject

class MainPresenter @Inject constructor(private val appApi: AppApi, private val context: Context) :
        RxPresenter<MainContract.View, ActivityMainBinding>(), MainContract.Presenter {

    fun toInit() {
        mContentView.setMainModel(MainModel())
        mContentView.userInfo = UserInfo("马齐", "18", "188")

    }

    inner class MainModel

}
